pubsub/cli.rb

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env ruby
+# Copyright 2015 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "gcloud"
+
+# Shows using gcloud to interact with Pub/Sub
+class PubSubSample
+  attr_accessor :topic, :sub
+
+  def initialize project
+    @project = project
+    @topic = "ruby-sample"
+    @sub = "ruby-sample-sub"
+    @subp = "ruby-sample-sub-p"
+
+    gcloud = Gcloud.new @project
+    pubsub = gcloud.pubsub
+
+    topic = pubsub.topic @topic, autocreate: true
+
+    sub = topic.get_subscription @sub
+    topic.subscribe @sub if sub.nil?
+
+    subp = topic.get_subscription @subp
+    topic.subscribe(
+      @subp,
+      endpoint: "https://#{@project}.appspot.com/push") if subp.nil?
+  end
+
+  def publish message
+    gcloud = Gcloud.new @project
+    pubsub = gcloud.pubsub
+    topic = pubsub.topic @topic
+    topic.publish message
+  end
+
+  def pull
```

Comment:
Hmm.  I kind of like the use of `@instance` variables.  If we use variables, though, let's pass them as method parameters so the user can see where they might come from instead of seeing a 'magic' instance variable.

```ruby
def pull project, subscription_name
  gcloud       = Gcloud.new project
  subscription = gcloud.subscription subscription_name

  subscription.pull.each do |message|
    message.acknowledge!
    puts message.data
  end
end
```